google-cloud-spanner/src/test/java/com/google/cloud/spanner/RetryOnInvalidatedSessionTest.java

```diff
@@ -217,17 +218,21 @@ public void setUp() throws InterruptedException {
       }
       // This prevents repeated retries for a large number of sessions in the pool.
       builder.setMinSessions(1);
+      SessionPoolOptions sessionPoolOptions = builder.build();
       spanner =
           SpannerOptions.newBuilder()
               .setProjectId("[PROJECT]")
               .setChannelProvider(channelProvider)
-              .setSessionPoolOption(builder.build())
+              .setSessionPoolOption(sessionPoolOptions)
               .setCredentials(NoCredentials.getInstance())
               .build()
               .getService();
       client = spanner.getDatabaseClient(DatabaseId.of("[PROJECT]", "[INSTANCE]", "[DATABASE]"));
       invalidateSessionPool(client, spanner.getOptions().getSessionPoolOptions().getMinSessions());
     }
+    assumeFalse(
```

Comment:
Note: I also have a large update (cleanup) of this test pending in https://github.com/googleapis/java-spanner/pull/3040.

Once this PR has been merged, I'll rebase my PR and update it so it executes this test for the cases that still use regular sessions, even when multiplexed sessions are enabled (i.e. for read/write transactions). That will ensure that the session retry continues to work for those cases as well after enabling multiplexed sessions.

Reply:
Sure, sounds good.